### DISCUSSION TITLE
Bug? Reusing ID's in variants after option deletion

### DISCUSSION THREAD

**Discussion by JoeMatkin** [swellstores/community#318] [Help] [answered]
Reusing ID's in variants after option deletion.

So whenever we delete a variant option those variants that get deleted pass there ID's and sometimes other data over to other variants, why is this?

**Comment by ericingram** [comment, marked as answer]
That does sound like it could be a bug. The only expected behavior around variant IDs is when one has been used in an order, it will archive the variant instead of deleting it outright, and restore it if one ends up getting re-created with the same options. Depending on if you delete it by deleting the option value itself or via API

cc @logeshswell

**Reply by JoeMatkin** [reply]
Hi @ericingram thanks for the reply;

Turns out the issue was because we where only sending the option_name to Swell when deleting an option, instead of sending both the option name and ID.

This may be useful to add to the docs as I don't think it mentions this, unless we missed it.

Thanks

**Reply by JoeMatkin** [reply]
Also I deleted an option which caused variant deletion, I then added the option back to restore the variant and the fields where not restored as you said they should be through the archive feature.

Don't know what's going on with this

**Reply by ericingram** [reply]
With the deleted option causing variant deletion, did that variant have any association with an existing order? That's the mechanism it should use to archive vs delete

**Reply by logeshswell** [reply]
Another possible reason could be whether the variant is associated with stock history. If so, variant will be archived instead of being deleted.

**Reply by JoeMatkin** [reply]
Ah this explains it, thanks
